This patch disable the ip tools options of the busybox config when the iproute2 package is selected
in buildroot config. It modify the makefile of the busybox package in buildroot.
We need to disable ip from busybox if we choose to use iproute2 because by default,
the binary used for ip is the ip of busybox (/bin/ip) and we need to use ip of iproute2
(/sbin/ip). This patch fix the conflict with the two binaries.

Signed-off-by: Kevin Joly <joly.kevin25@gmail.com>

Index: buildroot/package/busybox/busybox.mk
===================================================================
--- buildroot.orig/package/busybox/busybox.mk
+++ buildroot/package/busybox/busybox.mk
@@ -190,6 +190,15 @@ define BUSYBOX_NOCLOBBER_INSTALL
	$(SED) 's/^noclobber="0"$$/noclobber="1"/' $(@D)/applets/install.sh
 endef

+# If IPROUTE2 is selected, disable all the ip tools of busybox
+ifeq ($(BR2_PACKAGE_IPROUTE2),y)
+define BUSYBOX_IPROUTE2
+       $(call KCONFIG_DISABLE_OPT,CONFIG_FEATURE_IFUPDOWN_IP,$(BUSYBOX_BUILD_CONFIG))
+       $(call KCONFIG_DISABLE_OPT,CONFIG_FEATURE_IFUPDOWN_IFCONFIG_BUILTIN,$(BUSYBOX_BUILD_CONFIG))
+       $(call KCONFIG_DISABLE_OPT,CONFIG_IP,$(BUSYBOX_BUILD_CONFIG))
+endef
+endif
+
 define BUSYBOX_KCONFIG_FIXUP_CMDS
	$(BUSYBOX_SET_MMU)
	$(BUSYBOX_SET_LARGEFILE)
@@ -199,6 +208,7 @@ define BUSYBOX_KCONFIG_FIXUP_CMDS
	$(BUSYBOX_SET_CRYPT_SHA)
	$(BUSYBOX_NETKITBASE)
	$(BUSYBOX_NETKITTELNET)
+	$(BUSYBOX_IPROUTE2)
	$(BUSYBOX_INTERNAL_SHADOW_PASSWORDS)
	$(BUSYBOX_SET_INIT)
	$(BUSYBOX_SET_WATCHDOG)
